tests/CallingConvention.hs

```diff
@@ -65,6 +65,9 @@ copyPushR v = let pv = toPush v in pv ++ pv
 complexWhileCondR :: Data Int32 -> (Data Int32, Data Int32)
 complexWhileCondR y = whileLoop (0,y) (\(a,b) -> ((\a b -> a * a < b * b) a (b-a))) (\(a,b) -> (a+1,b))
 
+deepArrayCopyTest :: Data [[[Length]]] -> (Data [[[Length]]], Data [[[Length]]])
```

Comment:
The change we're now testing went in without changing the gold files for a single test which in hindsight looks a bit strange, shouldn't this also be a gold test in RegressionTests so we catch changes that implicitly affect array lowering?

Reply:
Seems reasonable. Will do.